### DISCUSSION TITLE
Active state not working

### DISCUSSION THREAD

**Discussion by Pedropfuenmayor** [tailwindlabs/tailwindcss#10444] [Help]
Hi everybody,

The active state is not working for me and i think is also not working in the docs page.

I am using version `^3.2.4` with NX Monorepo.

Here my  `tailwind.config.js`

```js
const { createGlobPatternsForDependencies } = require('@nrwl/next/tailwind');
const { join } = require('path');

module.exports = {
  content: [
    join(
      __dirname,
      '{src,pages,components,app}/**/*!(*.stories|*.spec).{ts,tsx,html}'
    ),
    ...createGlobPatternsForDependencies(__dirname),
  ],
  darkMode: 'class',
  theme: {
    extend: {
      colors: {
        dark: 'rgb(var(--color-dark) / <alpha-value>)',
        grey: 'rgb(var(--color-grey) / <alpha-value>)',
        lightGrey: 'rgb(var(--color-lightGrey) / <alpha-value>)',
        white: 'rgb(var(--color-white) / <alpha-value>)',
        primary: 'rgb(var(--color-primary) / <alpha-value>)',
      },
      opacity: {
        8: '0.08',
        12: '0.12',
        16: '0.16',
        24: '0.24',
        32: '0.32',
        48: '0.48',
        56: '0.56',
        64: '0.64',
        72: '0.72',
        88: '0.88',
        96: '0.96',
      },
    },
  },
  variants: {
    extend: {
      backgroundColor: ['active'],
    },
  },
  plugins: [],
};
```

and here my component (all the other modifiers/states work for fine):

```ts
import { cva } from 'class-variance-authority';

import { IconButtonProps, IconButtonSize } from './icon-button.interface';

export const buttonStyle = cva('flex items-center justify-center', {
  variants: {
    size: {
      small: 'h-6 w-6 rounded-lg',
      normal: 'h-8 w-8 rounded-lg',
      large: 'h-11 w-11 rounded-[48px]',
    },
    disabled: {
      true: 'cursor-not-allowed',
      false:
        'hover:bg-lightGrey hover:bg-opacity-16 active:bg-lightGrey active:bg-opacity-32 focus-visible:outline-none focus-visible:border focus-visible:border-[1.5px] focus-visible:border-primary focus-visible:border-opacity-80',
    },
  },

  defaultVariants: {
    size: IconButtonSize.NORMAL,
  },
});

const iconStyle = cva('', {
  variants: {
    size: {
      small: 'h-2 w-2 ',
      normal: 'h-3 w-3 ',
      large: 'h-4 w-4 ',
    },
    disabled: {
      true: 'text-dark text-opacity-60',
      false: 'text-dark',
    },
  },

  defaultVariants: {
    size: IconButtonSize.NORMAL,
  },
});

export function IconButton({
  Icon,
  disabled,
  size = IconButtonSize.NORMAL,
  ...props
}: IconButtonProps) {
  return (
    <button className={buttonStyle({ size, disabled })} {...props}>
      <Icon className={iconStyle({ size, disabled })} />
    </button>
  );
}

export default IconButton;
```

**Comment by wongjn** [comment]
Seems to working fine in a [Tailwind Play](https://play.tailwindcss.com/90E1pozETa) so it seems like it is something more particular to your project. Consider providing a remotely-hosted git repository such that we can help troubleshoot further.

**Reply by Pedropfuenmayor** [reply]
Thank you for the answer. Here a video of your tailwind play. I see that the hover state work, but when i click it does not keep the corresponding bg to the active state which is the problem i have. Am i understanding the active state functionality right?

https://www.loom.com/share/6cc761dc6f0e43198671f07081165c66

**Reply by wongjn** [reply]
I don't have an Mac OS device to test, so does the button change on click if you change the `active:bg-*` class to a completely different color?

**Reply by wongjn** [reply]
> In the playground i change the background color on the active state to: `active:bg-blue-900` and it doesn't work[^1]

Seems like something on your system is preventing the `:active` CSS pseudo class from working perhaps.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/10444#discussioncomment-4816850

**Comment by Pedropfuenmayor** [comment]
In the playground i change the background color on the active state to: `active:bg-blue-900` and it doesn't work

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/10444#discussioncomment-4817054 to keep the reply chain together.
